test/app-luatest/gh_7479_bug_fix_uri_lib_test.lua

```diff
@@ -0,0 +1,49 @@
+local t = require('luatest')
+local g = t.group()
+local popen   = require('popen')
+local net_box = require('net.box')
+local server  = require('test.luatest_helpers.server')
+
+-- Create test instance.
+g.before_all = function()
+    g.server = server:new({alias = 'test-gh-7479'})
+    g.server:start()
+    -- Get port for connection testing.
+    g.server_port = g.server:eval("return require('console').listen(0):name()['port']")
+end
+
+-- Stop test instance.
+g.after_all = function()
+    g.server:stop()
+end
+
+-- Checks whether it is possible to specify only port for tarantoolctl connect.
+g.test_tarantoolctl_connect = function()
+    local tarantoolctl_location = io.popen('which tarantoolctl'):read()
+    local cmd_tarantoolctl = {
+        tarantoolctl_location,
+        'connect',
+        -- Indicate only port, without localhost:* before the port value.
+        tostring(g.server_port),
+    }
+
+    -- Connection will be established in a separate process.
+    local ph_tarantoolctl = popen.new(cmd_tarantoolctl, {
+        stdin  = 'devnull',
+        stdout = 'devnull',
+        stderr = popen.opts.PIPE,
+    })
+    -- Stderr should contain a message about successful connection.
+    local result = ph_tarantoolctl:read({stderr = true}):rstrip()
+    ph_tarantoolctl:close()
+
+    t.assert_equals(result, 'connected to localhost:' .. tostring(g.server_port))
+end
+
+-- Checks whether it is possible to specify only port for net.box.connect.
+g.test_net_box_connect_with_only_port_indication = function()
+    -- Indicate only port, without localhost:* before the port value.
+    local ok, msg = pcall(net_box.connect, nil, g.server_port)
+    t.assert_equals(ok, true)
+    t.assert_is_not(tostring(msg), 'usage: connect(uri[, opts] | host, port[, opts])')
```

Comment:
No need to check the error message - the error code is enough. (We may change the error message in future, in which case the check would become pointless.)

Reply:
Done